From: Christian Kastner <ckk@kvr.at>
Date: Tue, 1 Sep 2015 15:43:17 +0200
Subject: Make testcase aware of a kernel behaviour change

This is analogous to the following commit:

	commit 7963a21a183ccc5658acff3a1bf05ec3b0688946
	Author: David Howells <dhowells@redhat.com>
	Date: Wed, 15 Jan 2014 15:07:05 +0000
	Subject: TEST: Make search/valid aware that the behaviour of the kernel changed

The kernel changed referred to also affected timeout/valid, which had
previously been affected by another change. All in all, between 3.8 and 3.12,
kernel returned ENOKEY instead of EKEYEXPIRED.

Bug-Debian: https://bugs.debian.org/768905
Applied-Upstream: https://git.kernel.org/cgit/linux/kernel/git/dhowells/keyutils.git/commit/?id=7a75d5690503ffb9eadf7f01fcc1a44e88994e33
Last-Update: 2015-09-30
---
 tests/keyctl/timeout/valid/runtest.sh | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/tests/keyctl/timeout/valid/runtest.sh b/tests/keyctl/timeout/valid/runtest.sh
index 0674826..e37ddc6 100644
--- a/tests/keyctl/timeout/valid/runtest.sh
+++ b/tests/keyctl/timeout/valid/runtest.sh
@@ -52,7 +52,13 @@ sleep_at_least 2
 # check the key has expired
 marker "CHECK NO READ PAYLOAD"
 print_key --fail $keyid
-expect_error EKEYEXPIRED
+kver=`uname -r`
+if ! version_less_than "$kver" 3.8 && version_less_than "$kver" 3.13
+then
+	expect_error ENOKEY
+else
+	expect_error EKEYEXPIRED
+fi
 
 # check revocation doesn't work
 marker "CHECK NO REVOKE KEY"
@@ -101,7 +107,13 @@ sleep_at_least 1
 # listing the session keyring should fail
 marker "CHECK NO LIST SESSION KEYRING"
 list_keyring --fail $keyringid
-expect_error EKEYEXPIRED
+kver=`uname -r`
+if ! version_less_than "$kver" 3.8 && version_less_than "$kver" 3.13
+then
+	expect_error ENOKEY
+else
+	expect_error EKEYEXPIRED
+fi
 
 # validating the new keyring's name and type should also fail
 marker "CHECK NO VALIDATE KEYRING"
